contracts/libraries/Samples.sol

```diff
@@ -113,6 +122,11 @@ library Samples {
                 params.secondsAgo[i]
             );
         }
+        averageTick = int24((tickSecondsAccum[secondsAgoLength - 1] - tickSecondsAccum[0]) 
+                                / int32(params.secondsAgo[secondsAgoLength - 1] - params.secondsAgo[0]));
+        averagePrice = TickMath.getSqrtRatioAtTick(averageTick);
```

Comment:
Here we calculate the average price based on the TWAP tick.

The TWAP tick is calculated using first and last indicies of `secondsAgo`.